keras/activations.py

```diff
@@ -78,6 +80,11 @@ def get(identifier):
         identifier = str(identifier)
         return deserialize(identifier)
     elif callable(identifier):
+        if isinstance(identifier, Layer):
+            warnings.warn((
+                'Never pass {identifier} into Activation() because '
+                '{identifier} itself is a layer.'
```

Comment:
This will need a more user-friendly error, e.g. "Do not pass a layer instance (such as {identifier}) as the `activation` argument of another layer. Instead, advanced activation layers should be used just like any other layer in a model."